src/applications/personalization/appointments/actions/index.js

```diff
@@ -1,27 +1,205 @@
+          `/vaos/v0/appointments?start_date=${startOfToday}&type=va`,
+        );
+        ccAppointments = await apiRequest(
+          `/vaos/v0/appointments?start_date=${startOfToday}&type=cc`,
+        );
+      }
+      const facilityIDs = uniq(
+        map(
+          vaAppointments,
+          appointment =>
+            getStagingID(appointment?.attributes?.sta6aid)
+              ? `vha_${getStagingID(appointment?.attributes?.sta6aid)}`
+              : undefined,
+        ),
+      );
+
+      if (environment.isLocalhost()) {
+        facilitiesResponse = MOCK_FACILITIES;
+      } else {
+        facilitiesResponse = await apiRequest(
+          `${environment.API_URL}/v1/facilities/va?ids=${facilityIDs.join(
+            ',',
+          )}`,
+        );
+      }
+
+      facilitiesLookup = reduce(
+        facilitiesResponse?.data,
+        (lookup, facility) => {
+          const facilityID = replace(facility?.id, 'vha_', '');
+          // eslint-disable-next-line no-param-reassign
+          lookup[facilityID] = facility;
+          return lookup;
+        },
+        {},
+      );
+    } catch (error) {
+      dispatch({
+        type: FETCH_CONFIRMED_FUTURE_APPOINTMENTS_FAILED,
       });
+    }
+
+    const formattedVAAppointments = reduce(
+      vaAppointments,
+      (accumulator, appointment) => {
+        const startDate = moment(appointment?.attributes?.startDate);
+        const now = moment();
+
+        // Past appointments should be filtered out already on the api side, this is a safe guard.
+        if (startDate.isBefore(now)) {
+          return accumulator;
+        }
+
+        // Derive the facility.
+        const facility =
+          facilitiesLookup[getStagingID(appointment?.attributes?.facilityId)];
+
+        accumulator.push({
+          additionalInfo: getAdditionalInfo(appointment),
+          facility,
+          id: appointment?.id,
+          isVideo: !isEmpty(appointment?.attributes?.vvsAppointments),
+          providerName: facility?.attributes?.name,
+          startsAt: startDate.toISOString(),
+          type: 'va',
+          timeZone: getTimeZone(
+            getStagingID(appointment?.attributes?.facilityId),
+          ),
+        });
+
+        return accumulator;
+      },
+      [],
+    );
+
+    const formattedCCAppointments = reduce(
+      ccAppointments,
+      (accumulator, appointment) => {
+        const startDate = moment(appointment?.attributes?.appointmentTime);
+        const now = moment();
+
+        // Escape early if the appointment is in the past.
+        if (startDate.isBefore(now)) {
+          return accumulator;
+        }
+
+        accumulator.push({
+          additionalInfo: getAdditionalInfo(appointment),
+          id: appointment?.id,
+          isVideo: false,
+          providerName: appointment?.attributes?.providerPractice,
+          startsAt: startDate.toISOString(),
+          timeZone: appointment?.attributes?.timeZone,
```

Comment:
TODO: This needs to be formatted differently. I asked Jeff for assistance and will update once clear.